File: python/leetcode007.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
            i = -1
            x = -x
        else:
            i = 1

        res = 0
        while x != 0:
            t = x % 10
            x = x // 10
            res = res*10 + t
        if i*res > -2**31 and i*res<2**31-1:
            return i*res
        else:
            return 0

if __name__ == "__main__":
    s = Solution()
    print(s.reverse(1463847412))

File: leetcodec++/leetcode007.cpp
//
// Created by 黄敏辉 on 2019-04-25.
//
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;


class Solution {
public:
    int reverse(int x) {
        long int res = 0;

        while (x != 0){
            res = res * 10 + x%10;
            x = x / 10;
            if(res > INT_MAX)
                return 0;
            if(res <INT_MIN)
                return 0;
        }
        return res;
    }
};